.github/workflows/nightly-build.yaml

```diff
@@ -0,0 +1,24 @@
+name: Nightly Build
+
+on:
```

Comment:
Can we also add [`workflow_dispatch`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch) to allow manual override when required?